### DISCUSSION TITLE
v2.8.0

### DISCUSSION THREAD

**Discussion by mienaiyami** [mienaiyami/yomikiru#73] [General]
## Downloads

[![Download setup](https://img.shields.io/badge/Windows%20Setup%20(exe)-Yomikiru--2.8.0--Setup.exe-brightgreen?logo=windows&logoColor=blue)](https://github.com/mienaiyami/yomikiru/releases/download/v2.8.0/Yomikiru-2.8.0-Setup.exe)

[![Download Portable](https://img.shields.io/badge/Windows%20Portable%20(zip)-Yomikiru--win32--2.8.0--Portable.zip-brightgreen?logo=windows&logoColor=blue)](https://github.com/mienaiyami/yomikiru/releases/download/v2.8.0/Yomikiru-win32-2.8.0-Portable.zip)

---
## Note

- App name will be changing to "Yomikiru".
- **Change in App state management system, some issues expected, please report.**

## Added

- Open item from location list in reader directly by double clicking (enable from settings).
- Better sorting for archive files as well.
- Options to reset settings and theme.
- Watch for file change; update bookmark and history in other instances of app as well.
- dev:
  - remove old shortcuts from settings.

## Changed

- Disabled scrollToPage on zen mode toggle when window is maximized.
- dev:
  - state management tool changed from context to redux for performance.
  - moved theme setting to themes file.

## Fixed

- Archive files not sorting in right order.

---

#### Todo

- ~~Better state management.~~
- Add ability to change image quality in canvas mode for less RAM usage and performance.
- Write tests.
- Adjust app for debian.

---

**Full Changelog**: https://github.com/mienaiyami/yomikiru/compare/v2.7.2...v2.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mienaiyami/yomikiru/releases/tag/v2.8.0'>v2.8.0</a>.</em>

**Comment by omodia** [comment]
> state management tool changed from context to redux.
What does this do?

**Reply by mienaiyami** [reply]
Basically, a lot better performance and code quality for better developer experience.

**Comment by omodia** [comment]
The theme maker is not correctly assigned to some options and some options are not working properly.

**Reply by mienaiyami** [reply]
Please make new issue showing where the bug is.

**Comment by omodia** [comment]
The history limit is not present, will it affect the performance / storage?
Edit: I am talking about the 'list item already read color' and not the 'continue reading' tab.

**Reply by mienaiyami** [reply]
It will not. Old system used to store bigger data for just one chapter. Now only manga is stored with chapter read name only.
It will not be any issue unless you have over 60 manga with 100+ chapter read in history.

**Reply by omodia** [reply]
I have over 100 comics, mostly all of them have 100+ chapters and many of them have 200+ chapters and some are 1000+ chapters.
That is why I am afraid of this method (list item already read color).

**Reply by mienaiyami** [reply]
> this method (list item already read color).

It has been there since earlier version; but it had some styling error and i had totally forgotten about it.

The example above was just an example. 
History is not changed frequently. And even if you did manage to read 100 comics with 1000 chapters, total size of data wont be any bigger than 1-3mb. There will not be any performance issue. If you do encounter one then i will limit "chapter read". Though its unlikely, that's why i removed history limit option.

**Comment by omodia** [comment]
Keeping the continue reading UI as it is and removing the 'list item already read color' record solve the issue.
Then only the current read chapter would be recorded.
Will this solve the issue?

**Reply by mienaiyami** [reply]
Create a new issue to track request, I will add option to disable chapter recording. But as i said above, it will not affect performance, only removes the ability to see "chapters read".
